src/hooks/useActiveAccount.tsx

```diff
@@ -56,28 +57,28 @@ export const ActiveAccountContextProvider = ({
     if (activeAccount?.account?.id) {
       return;
     }
-    const accountsWithProduct = filterAccounts(accounts.data);
-    if (accountsWithProduct.length < 1) {
+    const filteredAccounts = filterAccounts(accountsResult.data);
```

Comment:
The logic in this hook looks very similar to the logic in `setActiveAccountId` I wonder if we could move the `useEffect` below `setActiveAccountId` and call the callback function instead.